### PR TITLE
Clarify Promise.prototype.finally()

### PR DESCRIPTION
**Pull request by allan-bonadio**
#### Summary
Made examples 2 & 3 more clear.
Found a number of other small things that could be improved or clarified.

#### Motivation
- First sentence described a detail rather than help with the main description.
- clarified syntax code
- clarified that finally() returns an identical promise
- removed some text that was just confusing

#### Metadata
- [ ] Adds a new document
- [ x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ browser-compat: javascript.builtins.Promise.finally
 ---
 {{JSRef}}
 
-The **`finally()`** method returns a {{jsxref("Promise")}}.
+The **`finally()`** method is called on a {{jsxref("Promise")}}.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:16]:
```suggestion
The **`finally()`** method of a {{jsxref("Promise")}} schedules a function, the _callback function_, to be called when the promise is settled. Like `then()` and `catch()`, it immediately returns an equivalent {{jsxref("Promise")}} object, allowing to chain calls to a promise methods, an operation called _composition.
```

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 2]
```diff
@@ -28,8 +28,8 @@ This helps to avoid duplicating code in both the promise's {{jsxref("Promise.the
 ```js
 p.finally(onFinally);
 
-p.finally(function() {
-   // settled (fulfilled or rejected)
+p.finally(function onFinally() {
+   // will run after p is settled (fulfilled or rejected)
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:32]:
```suggestion
   // Code that will run after p is settled (fulfilled or rejected)
```

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 3]
```diff
@@ -40,8 +40,9 @@ p.finally(function() {
 
 ### Return value
 
-Returns a {{jsxref("Promise")}} whose `finally` handler is set to the
-specified function, `onFinally`.
+Passes through an equivalent {{jsxref("Promise")}} to the one this call was made on.
+If the function callback throws an error, the return promise will be rejected 
+with that value instead.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:45]: This change is incorrect I think. What about
```suggestion
Returns an equivalent {{jsxref("Promise")}} with its `finally` handler set to the specified function.
```

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 4]
```diff
@@ -53,22 +54,23 @@ The `finally()` method is very similar to calling
...
-  - Unlike `Promise.resolve(2).then(() => {}, () => {})` (which
-    will be resolved with `undefined`),
-    `Promise.resolve(2).finally(() => {})` will be resolved with
-    `2`.
-  - Similarly, unlike `Promise.reject(3).then(() => {}, () => {})`
-    (which will be fulfilled with `undefined`),
-    `Promise.reject(3).finally(() => {})` will be rejected with
-    `3`.
+  fulfillment value, and so there's no need to provide it. 
+- A `finally` call will chain through an equivalent to the orignal promise.
```
Comment (teoli2003) [inline files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:60]:
```suggestion
- A `finally` call will chain through an equivalent to the original promise.  
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
This is already very good. I added some suggestions/corrections.

**Review by OnkarRuikar — COMMENTED** [review]
Forgot to to click complete. Ignore if not relevant anymore.

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 5]
```diff
@@ -28,8 +28,8 @@ This helps to avoid duplicating code in both the promise's {{jsxref("Promise.the
 ```js
 p.finally(onFinally);
 
-p.finally(function() {
-   // settled (fulfilled or rejected)
+p.finally(function onFinally() {
```
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:31]: I went with the arrow function

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md (hunk 1 above)
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:16]: changed to almost the same

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md (hunk 2 above)
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:32]: I adopted that wording, better

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md (hunk 3 above)
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:45]: done

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md (hunk 4 above)
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md]: doh!  done

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 6]
```diff
@@ -28,8 +28,8 @@ This helps to avoid duplicating code in both the promise's {{jsxref("Promise.the
 ```js
 p.finally(onFinally);
```
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:29]: done

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:31]: I changed it to an arrow function as per suggestion above

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:32]: ... will run after promise is settled...

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 7]
```diff
@@ -53,22 +54,23 @@ The `finally()` method is very similar to calling
...
-    `2`.
-  - Similarly, unlike `Promise.reject(3).then(() => {}, () => {})`
-    (which will be fulfilled with `undefined`),
-    `Promise.reject(3).finally(() => {})` will be rejected with
-    `3`.
+  fulfillment value, and so there's no need to provide it. 
+- A `finally` call will chain through an equivalent to the orignal promise.  
+  So for example:
+  - Unlike `Promise.resolve(2).then(() => 77, () => {})` (which
+    will return a promise which will be resolved with `77`),
```
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:63]: yeah, that's better wording.  used it.

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 8]
```diff
@@ -53,22 +54,23 @@ The `finally()` method is very similar to calling
...
-    (which will be fulfilled with `undefined`),
-    `Promise.reject(3).finally(() => {})` will be rejected with
-    `3`.
+  fulfillment value, and so there's no need to provide it. 
+- A `finally` call will chain through an equivalent to the orignal promise.  
+  So for example:
+  - Unlike `Promise.resolve(2).then(() => 77, () => {})` (which
+    will return a promise which will be resolved with `77`), 
+    `Promise.resolve(2).finally(() => {})` will return a 
+    promise which will be resolved with `2`.
```
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:65]: done

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 9]
```diff
@@ -53,22 +54,23 @@ The `finally()` method is very similar to calling
...
+  - Similarly, unlike `Promise.reject(3).then(() => {}, () => 88)`
+    (which will, again, return a promise which will be rejected with `88`),
```
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md]: done.  It's good I'm learning the lingo.

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 10]
```diff
@@ -53,22 +54,23 @@ The `finally()` method is very similar to calling
...
+    `Promise.reject(3).finally(() => {})` will return a promise
```
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md]: done

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 11]
```diff
@@ -53,22 +54,23 @@ The `finally()` method is very similar to calling
...
+    `Promise.resolve(2).finally(() => {})` will return a 
+    promise which will be resolved with `2`.
+  - Similarly, unlike `Promise.reject(3).then(() => {}, () => 88)`
+    (which will, again, return a promise which will be rejected with `88`),
+    `Promise.reject(3).finally(() => {})` will return a promise 
+    which will be rejected with `3`.
 
 > **Note:** A `throw` (or returning a rejected promise) in the
-> `finally` callback will reject the new promise with the rejection reason
+> `finally` callback will reject the promise, which will, in that case,
```
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:75]: will reject the returned promise with the reason `99`.

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 12]
```diff
@@ -53,22 +54,23 @@ The `finally()` method is very similar to calling
...
+- A `finally` call will chain through an equivalent to the orignal promise.  
+  So for example:
+  - Unlike `Promise.resolve(2).then(() => 77, () => {})` (which
+    will return a promise which will be resolved with `77`), 
+    `Promise.resolve(2).finally(() => {})` will return a 
+    promise which will be resolved with `2`.
+  - Similarly, unlike `Promise.reject(3).then(() => {}, () => 88)`
+    (which will, again, return a promise which will be rejected with `88`),
+    `Promise.reject(3).finally(() => {})` will return a promise 
+    which will be rejected with `3`.
```
[the remark this answers is not in the document]
Comment (allan-bonadio) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md]: yup

**Comment by teoli2003** [comment]
Hi! Thanks for updating the PR. I don't exactly what's going on, but I don't see your changes. Have you pushed them here?

**Comment by allan-bonadio** [comment]
> Hi! Thanks for updating the PR. I don't exactly what's going on, but I don't see your changes. Have you pushed them here?

sorry, never did a PR on github before.  Too many buttons and distractions, kindof confusing.  I updated the wrong branch.  Now it should be right.

**Review by teoli2003 — APPROVED** [review]
No worry, Git and GitHub feel different at the beginning, but work well once we are used to them.

This looks good to me. If we find more changes to be done, we can do this in follow-up PRs.

This is much better.

**Comment by teoli2003** [comment]
\o/

**Pull request merged**